packages/shared/src/components/fields/CustomSwitch.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ReactElement, useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import styles from './CustomSwitch.module.css';
+import classed from '../../lib/classed';
+
+export interface ContentsSwitchProps {
+  className?: string;
+  inputId: string;
+  name: string;
+  checked?: boolean;
+  onToggle?: () => unknown;
+  leftContent: React.FC<{ className?: string }> | string;
+  rightContent: React.FC<{ className?: string }> | string;
+}
+
+const ContentContainer = classed(
+  'span',
+  'relative flex items-center justify-center z-2 px-3',
+);
+
+const baseContentClass = 'transform transition-transform text-base';
+const unselectedContentClass = 'text-theme-label-tertiary scale-125';
+const selectedContentClass = 'scale-150';
+
+export function CustomSwitch({
+  className,
+  inputId,
+  name,
+  checked,
+  onToggle,
+  leftContent: LeftContent,
+  rightContent: RightContent,
+}: ContentsSwitchProps): ReactElement {
+  const leftRef = useRef<HTMLElement>();
+  const rightRef = useRef<HTMLElement>();
+  const [width, setWidth] = useState('');
+
+  useEffect(() => {
+    if (!leftRef?.current || !rightRef?.current) {
+      return;
+    }
+
+    const element = checked ? rightRef.current : leftRef.current;
+    const { width: elementWidth } = element.getBoundingClientRect();
+    setWidth(`${elementWidth}px`);
+  }, [checked]);
+
+  const leftClasses = classNames(
+    baseContentClass,
+    checked ? unselectedContentClass : selectedContentClass,
+  );
+  const rightClasses = classNames(
+    baseContentClass,
+    checked ? selectedContentClass : unselectedContentClass,
+  );
+
+  return (
+    <label
+      className={classNames(
+        'group relative flex h-9 w-fit cursor-pointer typo-callout font-bold',
+        className,
+      )}
+      htmlFor={inputId}
+    >
+      <input
+        id={inputId}
+        name={name}
+        type="checkbox"
+        className="absolute w-0 h-0 opacity-0"
+        checked={checked}
+        onChange={onToggle}
+      />
+      <ContentContainer
+        ref={leftRef}
+        className={checked && 'text-theme-label-tertiary'}
+      >
+        {typeof LeftContent === 'string' ? (
```

Comment:
I see quite a lot of these typeof checks why do we need them? Is there anything else we render in here?

Reply:
This is to easily accommodate string kind of parameter. How about creating two different switch components to handle text only and the other side of this condition? Would love to hear what you think 🚢